src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -960,33 +931,34 @@ && nonNull(profileResponse.getIdamStatus())) {
         var hasNameChanged = !cwrProfileRequest.getFirstName().equals(userProfileResponse.getFirstName())
                 || !cwrProfileRequest.getLastName().equals(userProfileResponse.getLastName());
         if (isNotEmpty(mergedRoles) || hasNameChanged || !cwrProfileRequest.isStaffAdmin()) {
-            return updateMismatchedDatatoUP(cwrProfileRequest, idamId, mergedRoles, hasNameChanged);
+            return updateMismatchedDatatoUP(cwrProfileRequest, idamId, mergedRoles, hasNameChanged,
+                    userProfileResponse.getIdamStatus());
         }
 
         return true;
     }
 
     private boolean updateMismatchedDatatoUP(StaffProfileCreationRequest cwrProfileRequest, String idamId,
                                              Set<RoleName> mergedRoles,
-                                             boolean hasNameChanged) {
+                                             boolean hasNameChanged, String idamStatus) {
         UserProfileUpdatedData.UserProfileUpdatedDataBuilder builder = UserProfileUpdatedData.builder();
 
         if (isNotEmpty(mergedRoles)) {
-            builder
-                    .rolesAdd(mergedRoles);
+            builder.rolesAdd(mergedRoles);
         }
 
         if (!cwrProfileRequest.isStaffAdmin()) {
             builder.rolesDelete(Set.of(new RoleName(ROLE_STAFF_ADMIN)));
         }
 
         if (hasNameChanged) {
-
-            builder
-                    .firstName(cwrProfileRequest.getFirstName())
-                    .lastName(cwrProfileRequest.getLastName())
-                    .idamStatus(STATUS_ACTIVE);
-
+            builder.firstName(cwrProfileRequest.getFirstName())
+                    .lastName(cwrProfileRequest.getLastName());
+        }
+        if (!cwrProfileRequest.isSuspended()) {
```

Comment:
setting idam status in the UP request all the times even though there is no change in the status or first name or last name will eventually made 2 idam calls one is for role change(/api/v1/users/{userId}/roles) and second is for userDetails (/api/v1/users/{userId}) which is not necessary

Reply:
when there is no changes in the first name , lastname, status , only roles it will call modify roles in UP but not anything else ...